storage/memoized_iterator.go

```diff
@@ -85,13 +82,15 @@ func (b *MemoizedSeriesIterator) Seek(t int64) chunkenc.ValueType {
 		b.valueType = b.it.Seek(t0)
 		if b.valueType == chunkenc.ValNone {
 			return chunkenc.ValNone
+		} else if b.valueType == chunkenc.ValHistogram {
+			b.valueType = chunkenc.ValFloatHistogram
 		}
```

Comment:
We generally (and as of recently ;) prefer `switch` instead of `else if` constructs:

```suggestion
		switch b.ValueType {
			case chunkenc.ValNone:
				return chunkenc.ValNone
			case chunkenc.ValHistogram:
				b.valueType = chunkenc.ValFloatHistogram
		}
```